azurerm/resource_arm_virtual_hub.go

```diff
@@ -0,0 +1,410 @@
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Virtual Hub %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Virtual Hub %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if props := resp.VirtualHubProperties; props != nil {
+		d.Set("address_prefix", props.AddressPrefix)
+		if props.VirtualWan != nil {
+			if err := d.Set("virtual_wan_id", props.VirtualWan.ID); err != nil {
+				return fmt.Errorf("Error setting `virtual_wan_id`: %+v", err)
+			}
+		}
+		if props.VpnGateway != nil {
+			if err := d.Set("s2s_vpn_gateway_id", props.VpnGateway.ID); err != nil {
+				return fmt.Errorf("Error setting `s2s_vpn_gateway_id`: %+v", err)
+			}
+		}
+		if props.P2SVpnGateway != nil {
+			if err := d.Set("p2s_vpn_gateway_id", props.P2SVpnGateway.ID); err != nil {
+				return fmt.Errorf("Error setting `p2s_vpn_gateway_id`: %+v", err)
+			}
+		}
+		if props.ExpressRouteGateway != nil {
+			if err := d.Set("express_route_gateway_id", props.ExpressRouteGateway.ID); err != nil {
+				return fmt.Errorf("Error setting `express_route_gateway_id`: %+v", err)
+			}
+		}
+		if err := d.Set("virtual_network_connection", flattenArmVirtualHubVirtualNetworkConnection(props.VirtualNetworkConnections)); err != nil {
+			return fmt.Errorf("Error setting `virtual_network_connection`: %+v", err)
+		}
+		if props.RouteTable != nil {
+			if err := d.Set("route", flattenArmVirtualHubRoute(props.RouteTable.Routes)); err != nil {
+				return fmt.Errorf("Error setting `route`: %+v", err)
+			}
+		}
+	}
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceArmVirtualHubDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Network.VirtualHubClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["virtualHubs"]
+
+	future, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error deleting Virtual Hub %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		if !response.WasNotFound(future.Response()) {
+			return fmt.Errorf("Error waiting for deleting Virtual Hub %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+
+	return nil
+}
+
+func expandArmVirtualHubVirtualNetworkConnection(input []interface{}) *[]network.HubVirtualNetworkConnection {
+	if len(input) == 0 {
+		return nil
+	}
+
+	results := make([]network.HubVirtualNetworkConnection, 0)
+
+	for _, item := range input {
+		v := item.(map[string]interface{})
```

Comment:
We should confirm that this isn't nil before casting it.

Reply:
Fixed